api/v1alpha1/labels.go

```diff
@@ -8,8 +8,12 @@ const (
 	CredentialTypeLabelValueGit   = "git"
 	CredentialTypeLabelValueHelm  = "helm"
 	CredentialTypeLabelValueImage = "image"
+	CredentialTypeLabelGeneric    = "generic"
 
 	// Project Secrets
+	// Deprecated: Use CredentialTypeLabelGeneric instead. This label should not
+	// be used and won't be documented, but for now we plan to support it
+	// indefinitely for backwards compatibility.
```

Comment:
Is there a need to support this indefinitely instead of just phasing it out in 2 minor versions from now?

Reply:
I'd prefer that as well, although it doesn't feel especially onerous in this case to continue.

@jessesuen wdyt?

Reply:
Agree with Hidde. Would rather not keep it around too long

Reply:
Ok. Will adjust the comment accordingly.

Reply:
Fixed in 27c0c77f089062cde33780bb84efe37f6dba4e39.